lib/iris/analysis/maths.py

```diff
@@ -149,7 +149,7 @@ def _assert_matching_units(cube, other, operation_noun):
     if cube.units != getattr(other, 'units', cube.units):
         raise iris.exceptions.NotYetImplementedError(
             'Differing units (%s & %s) %s not implemented' %
-            (cube.units, other.units, operation_noun))
+            (cube.units, other.units, operation_name))
```

Comment:
I'm not sure this error message has been helped by this change:
```python
'Differing units (m & kg) add not implemented'
```

Maybe `Cannot %s differing units (%s & %s).format(operation_name, cube.units, other.units)`:

```python
'Cannot add differing units (m & kg)'
```

or something.